src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -83,9 +79,27 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
 
     @Override
     public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+        MatchAllQueryBuilder query = new MatchAllQueryBuilder();
+        query.fromXContent(parseContext);
+        return query.toQuery(parseContext);
+    }
+
+    public Query toQuery(QueryParseContext parseContext) {
+        if (boost == 1.0f) {
+            return Queries.newMatchAllQuery();
+        }
+
+        //LUCENE 4 UPGRADE norms field is not supported anymore need to find another way or drop the functionality
+        //MatchAllDocsQuery query = new MatchAllDocsQuery(normsField);
+        MatchAllDocsQuery query = new MatchAllDocsQuery();
```

Comment:
Going to drop this since it's not supported in lucene and there seems to be nothing on this option in the docs, nor is that field parsed anywhere.